frontend/src/components/pages/CampOverview/MoveCamperModal/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect } from "react";
+
+import { ModalOverlay, ModalContent, Modal } from "@chakra-ui/react";
+import { CampSession } from "../../../../types/CampsTypes";
+import { Camper } from "../../../../types/CamperTypes";
+import CamperAPIClient from "../../../../APIClients/CamperAPIClient";
+import PageOne from "./PageOne";
+import PageTwo from "./PageTwo";
+
+type MoveCamperModalProps = {
+  camper: Camper;
+  campSession: CampSession;
+  campSessions: CampSession[];
+  selectedCampSession: string;
+  setSelectedCampSession: (campSession: string) => void;
+  moveCampers: (camperIds: string[], campSessionId: string) => void;
+  moveCamperModalIsOpen: boolean;
+  moveCamperModalOnClose: () => void;
+};
+
+const MoveCamperModal = ({
+  camper,
+  campSession,
+  campSessions,
+  selectedCampSession,
+  setSelectedCampSession,
+  moveCampers,
+  moveCamperModalIsOpen,
+  moveCamperModalOnClose,
+}: MoveCamperModalProps): JSX.Element => {
+  const [modalPage, setModalPage] = React.useState<number>(1);
+  const [campers, setCampers] = React.useState<Camper[]>([]);
+  const [camperIsSelected, setCamperIsSelected] = React.useState<boolean[]>([]);
+
+  const reset = () => {
+    setModalPage(0);
+    setCampers([]);
+    setCamperIsSelected([]);
+  };
+
+  // Check if the camper has group members
+  useEffect(() => {
+    // Used to prevent the flash of previous state
+    reset();
+
+    // Check if the camper object is not null
+    if (Object.keys(camper).length !== 0) {
+      // Check if camper has group members
+      CamperAPIClient.getCampersByChargeId(camper.chargeId).then(
```

Comment:
Sorry about the confusion, this should be `getCampersByChargeIdAndSessionId`. We should only be able to move campers who were registered together and who are part of the current selected session. So you would pass in the current session id to the sessionID parameter.